BotController/BotController.cpp

```diff
@@ -1,12 +1,340 @@
-// BotController.cpp : This file contains the 'main' function. Program execution begins and ends there.
-//
-
 #include <iostream>
+#include <Winuser.h>
 #include "Robot.hpp"
+//include "RGBQuadValues.hpp"
 
 int main()
 {
-    std::cout << "Hello World!\n";
+    Robot robot = Robot();
+
+    //how exact the pixel color's RGBs need to be
+    double tolerance = 0.0;
+
+    //in milliseconds
+    DWORD clickSpeed = 80;
+    DWORD typeSpeed = 40;
+
+    //checking for new champ selects to avoid repeating actions
+    //set to false after finishing all champ select specific actions in a new champ select state
+    //set to true after accepting a new match
+    bool newChampSelect = true;
+
+    //avoid repeating starting actions in game
+    //set to false at the end of newGame if actions
+    //set to true after accepting a new match
+    bool newGame = true;
+
+    //avoid repeating the buy action until the base is left and re-entered
+    bool ableToBuyItems = true;
+
+    //bot running loop
+    while (true) {
+
+        //if entering a new game, do these things before normal in game actions
+        //actions: wait 5 seconds, buy items, attach to ally
+        if (newGame) {
```

Comment:
Could we try rewriting this so that it is something like
```cpp
enum BotStates {
    BotStartedState, NewGameState, InGameState, PostGameState, BotTerminatedState
};
```
```cpp
state = NewGameState;
do {
    switch(state) {
        case NewGameState:
            /* perform actions for new game */
            break;
        case InGameState:
            /* perform actions for in game */
            break;
        case PostGameState:
            /* perform actions for post game */
            break;
    }
} while (state != BotTerminatedState);
```
This formalizes the state machine in an `Enum`. The approach above uses a C-style enum, which is discouraged in C++, when compared to `enum class`. The problem with C++ enum class is that it requires explicit scoping which is annoying. We'll probably resort to:
```cpp
enum class BotState {
    /* states */
};
/* ... */
// State scope is implicit
using BotState;
```